src/main/scala/info/bethard/timenorm/TimeNormScorer.scala

```diff
@@ -15,6 +15,14 @@ object TimeNormScorer {
   case class Timex(id: String, textSpan: (Int, Int), time: TimeExpression) {
     val (textStart, textEnd) = textSpan
 
+    if (intervals(time).size == 0){
+      throw new IllegalStateException("Time expression without intervals")
+    }
+    for (interval <- intervals(time)) {
```

Comment:
You're trying to provoke an exception here, right? Can't you just check `interval.isDefined`? That, coupled with the comment above, should allow you to simply filter, rather than throwing exceptions and using a `Try`.